File: AngularFrontend/src/app/core/domain-classes/workflow.ts
import { WorkflowInstance } from './workflow-instance';
import { WorkflowStep } from './workflow-step';
import { WorkflowTransition } from './workflow-transition';

export class Workflow {
  id?: string;
  name: string;
  description: string;
  isWorkflowSetup: boolean;
  workflowSteps: WorkflowStep[];
  workflowTransitions: WorkflowTransition[];
  isWorkflowRedirectNextStep?: boolean;
  isWorkflowStepRedirectNextStep?: boolean;
  updatedAt?: Date;
  workflowInstances: WorkflowInstance[];
}

File: MySQL-NET-REST-API/DocumentManagement.Data/Entities/Workflow.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace DocumentManagement.Data
{
    public class Workflow : BaseEntity
    { 
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool IsWorkflowSetup { get; set; }
        public Guid UserId { get; set; }
        [ForeignKey("UserId")]
        public User User { get; set; }
        public ICollection<WorkflowStep> WorkflowSteps { get; set; }
        public ICollection<WorkflowInstance> WorkflowInstances { get; set; }
        public ICollection<WorkflowTransition> WorkflowTransitions { get; set; }
    }
}
